meadow-client/src/Marlowe/Gen.purs

```diff
@@ -0,0 +1,124 @@
+module Marlowe.Gen where
+
+import Prelude
+
+import Control.Lazy (class Lazy, defer)
+import Control.Monad.Gen (class MonadGen, chooseInt)
+import Control.Monad.Rec.Class (class MonadRec)
+import Data.BigInteger (BigInteger)
+import Data.Newtype (wrap)
+import Data.NonEmpty ((:|))
+import Semantics (Contract(..), IdChoice, Observation(..), Value(..))
+
+import Control.Monad.Gen as Gen
+import Data.BigInteger as BigInteger
+
+genBigInteger :: forall m. MonadGen m => MonadRec m => m BigInteger
+genBigInteger = BigInteger.fromInt <$> chooseInt (negate 2147483648) 2147483647
```

Comment:
We've got a `Bounded Int` instance to play with, so how about `BigInteger.fromInt <$> chooseInt bottom top`?